### PR TITLE
Partial implementation for simple_wait_frame that covers publisher confirms

### PR DESCRIPTION
**Pull request by b0max**
- Implementation covers ack, nack, return, deliver method frames, content header and content body frames.
- Unit tests and publisher confirm example

### CODE REVIEW

**Comment by b0max** [comment]
Hi Elias,

Its partial for simple_wait_frame function as it can return any type of
frames. Its complete for publisher confirms scenario. I don't have plan to
implement the rest of frames in simple_wait_frame at the moment. But if its
needed to move this PR forward I will do it.

Thanks

**Comment by lokedhs** [comment]
Thank you for the information. I don't mind merging this. The style is generally OK, so the only thing I would ask you to change is to make sure that class accessors follow the same naming convention as the rest of the code. I.e. the accessor for slot `name` in class `foo` should be called `foo/name`.

**Comment by b0max** [comment]
Hi Elias, I updated the code according to naming convention.

File: src/amqp.lisp [hunk 1]
```diff
@@ -159,11 +159,13 @@
       (t (error "Unexpected error: ~s" reply-type)))))
 
 (defun verify-rpc-framing-call (state channel result)
-  (when (cffi:null-pointer-p result)
-    (verify-rpc-reply state channel (amqp-get-rpc-reply state))))
+  (if (cffi:null-pointer-p result)
+      (verify-rpc-reply state channel (amqp-get-rpc-reply state))
+      t))
```
Comment (lokedhs) [inline src/amqp.lisp:164]: Why was the return value of `T` added here? This function is only ever called in the unwind handler of an `UNWIND-PROTECT` so the return value should not matter.

File: src/amqp.lisp [hunk 2]
```diff
@@ -159,11 +159,13 @@
...
 
 (defun maybe-release-buffers (state)
-  (amqp-maybe-release-buffers state))
+  (amqp-maybe-release-buffers state)
+  t)
```
Comment (lokedhs) [inline src/amqp.lisp:168]: The return value of `T` here has the same concern as the previous point. The return value is never used. Is there a reason for this change?

File: src/amqp.lisp [hunk 3]
```diff
@@ -845,3 +847,428 @@ retrieved has been processed"
...
+  (make-instance 'method-frame-basic-deliver
+                 :channel channel
+                 :consumer-tag consumer-tag
+                 :delivery-tag delivery-tag
+                 :redeliveredp redeliveredp
+                 :exchange exchange
+                 :routing-key routing-key))
+
+(defmethod print-object ((obj method-frame-basic-deliver) stream)
+  (print-unreadable-object (obj stream :type t :identity nil)
```
Comment (lokedhs) [inline src/amqp.lisp:1063]: If any of the slots are unbound when the object is printed, you might end up with an error while printing an object. This is particularly annoying when the debugger is printing it. There is a function in `misc.lisp` which handles this, please take a look at `PRINT-UNREADABLE-SAFELY`.

File: src/amqp.lisp [hunk 4]
```diff
@@ -845,3 +847,428 @@ retrieved has been processed"
...
+
+CHANNEL => INTEGER, Rabbitmq channel.
+BODY => ARRAY of BYTES, contents."
+  (make-instance 'content-body-frame
+                 :channel channel
+                 :body (bytes->array body)))
+
+(defmethod print-object ((obj content-body-frame) stream)
+  (print-unreadable-object (obj stream :type t :identity nil)
+  (format stream "CHANNEL ~s Body Size ~s"
```
Comment (lokedhs) [inline src/amqp.lisp:1140]: This seems incorrectly indented.

File: src/amqp.lisp [hunk 5]
```diff
@@ -845,3 +847,428 @@ retrieved has been processed"
...
+Arguments:
+
+METHOD-ID => INTEGER, AMQP basic return method id.
+METHOD-STRUCT => CFFI FOREIGN POINTER, decoded method frame.
+CHANNEL => INTEGER, Rabbitmq channel."
+  (declare (ignore method-struct channel))
+  (when (null (find-if #'(lambda (value)
+                           (eql var value))
+                       '(+amqp-confirm-select-ok-method+)))
+    (error "Unhandled method frame, ID ~s" method-id))
```
Comment (lokedhs) [inline src/amqp.lisp:1222]: I would suggest declaring a dedicated condition for this.

**Review by lokedhs — CHANGES_REQUESTED** [review]
Thank you for your work. I did notice a few issues. Would you mind looking at these?

File: src/amqp.lisp (hunk 1 above)
Comment (b0max) [reply src/amqp.lisp:164]: This is a mistake, I was trying something and forgot to revert it. sorry. reverted

File: src/amqp.lisp (hunk 2 above)
Comment (b0max) [reply src/amqp.lisp:168]: Same

File: src/amqp.lisp (hunk 3 above)
Comment (b0max) [reply src/amqp.lisp:1063]: Updated

File: src/amqp.lisp (hunk 4 above)
Comment (b0max) [reply src/amqp.lisp:1140]: fixed

File: src/amqp.lisp (hunk 5 above)
Comment (b0max) [reply src/amqp.lisp:1222]: Done

**Comment by b0max** [comment]
Thank you for your feedback, I updated the code.

**Comment by lokedhs** [comment]
Thank you for your contribution. The change has been merged.

**Pull request merged**